core/src/main/java/com/exadel/etoolbox/linkinspector/core/servlets/ReplaceByPatternServlet.java

```diff
@@ -194,53 +191,74 @@ private List<UpdatedItem> processResources(Collection<GridResource> gridResource
-                .filter(gridResource ->
-                        StringUtils.isNoneBlank(gridResource.getHref(), gridResource.getResourcePath(), gridResource.getPropertyName())
-                )
-                .filter(gridResource -> pattern.matcher(gridResource.getHref()).find())
-                .filter(gridResource ->
-                        repositoryHelper.hasReadWritePermissions(session, gridResource.getResourcePath()))
-                .limit(maxUpdatedItemsCount)
-                .collect(Collectors.toList());
+        return gridResources.stream().map(resources ->
+                resources.stream()
+                        .filter(gridResource ->
+                                StringUtils.isNoneBlank(gridResource.getHref(), gridResource.getResourcePath(), gridResource.getPropertyName())
+                        )
+                        .filter(gridResource -> pattern.matcher(gridResource.getHref()).find())
+                        .filter(gridResource ->
+                                repositoryHelper.hasReadWritePermissions(session, gridResource.getResourcePath()))
+                        .limit(maxUpdatedItemsCount)
+                        .collect(Collectors.toList())).collect(Collectors.toList());
     }
 
-    private List<UpdatedItem> replaceByPattern(Collection<GridResource> gridResources,
+    private List<UpdatedItem> replaceByPattern(List<List<GridResource>> gridResources,
                                                boolean isDryRun,
                                                String linkPattern,
                                                String replacement,
                                                ResourceResolver resourceResolver) throws PersistenceException {
         List<UpdatedItem> updatedItems = new ArrayList<>();
-        for (GridResource gridResource : gridResources) {
-            if (isDeactivated) {
-                break;
-            }
-            String currentLink = gridResource.getHref();
-            String path = gridResource.getResourcePath();
-            String propertyName = gridResource.getPropertyName();
-            Optional<String> updated = Optional.of(currentLink.replaceAll(linkPattern, replacement))
-                    .filter(updatedLink -> !updatedLink.equals(currentLink))
-                    .filter(updatedLink ->
-                            linkHelper.replaceLink(resourceResolver, path, propertyName, currentLink, updatedLink)
-                    );
-            if (updated.isPresent()) {
-                updatedItems.add(new UpdatedItem(currentLink, updated.get(), path, propertyName));
-                LOG.trace("The link was updated: location - {}@{}, currentLink - {}, updatedLink - {}",
-                        path, propertyName, currentLink, updated.get());
-                if (!isDryRun && updatedItems.size() % commitThreshold == 0) {
-                    resourceResolver.commit();
+        for (int i = 0; i < gridResources.size(); i++) {
```

Comment:
Technically, usually they forbid to run business logic inside servlet. Consider creating a service for business logic. Also, there might be a change to unify/reuse some code with the "FixBrokenLinks" servlet. Part of the code looks similar there. Also, log messages, etc